### PR TITLE
fix: Move execution of fee estimation within the Tx bounds

### PR DESCRIPTION
**Pull request by hansl**
This allows for isSigning to be set right away, before proceeding
to estimating fees, which can take a while. Thus every component
reacting to isSigning (and showing the three bouncing dots) will
be disabled right away, without waiting for server requests to
finish.

Also moved the validation schema for sendForm to its own schema
file and import. I did not want to touch ibcSendForm, but the
idea would be to reuse the schema there too. Will do in a follow
up PR.

Finally, fixed some internals with AmountInput and the way it
interacts with schemas and Formik. This new version returns only
actual number (or undefined) instead of a string that needs
parsing. This makes it easier to use in most cases too.

Fixes https://github.com/liftedinit/manifest-app/issues/308

## Summary by CodeRabbit

- **Refactor**
	- Deferred fee estimation across transaction flows for dynamic, on-demand fee calculation during transaction execution.
	- Streamlined transaction state management for improved responsiveness.

- **New Features**
	- Enhanced validations in transfer and mint forms, ensuring accurate recipient and amount inputs.
	- Improved numeric input handling to restrict invalid characters and provide a smoother user experience.
	- Introduced a new validation schema for forms, enhancing structure and maintainability.

- **Tests**
	- Updated numeric input tests to align with the new type and validation improvements.

### CODE REVIEW

File: components/bank/forms/ibcSendForm.tsx [hunk 1]
```diff
@@ -120,51 +119,14 @@ export default function IbcSendForm({
...
-    selectedToken: CombinedBalanceInfo;
-    memo: string;
-  }) => {
-    setIsSending(true);
+  const handleSend = async (values: sendForm.SendForm) => {
     try {
       // Convert amount to base units
       const exponent = values.selectedToken.metadata?.denom_units[1]?.exponent ?? 6;
-      const amountInBaseUnits = parseNumberToBigInt(values.amount, exponent).toString();
+      const amountInBaseUnits = parseNumberToBigInt('' + values.amount, exponent).toString();
```
Comment (fmorency) [inline components/bank/forms/ibcSendForm.tsx:129]:
```suggestion
      const amountInBaseUnits = parseNumberToBigInt(values.amount.toString(), exponent).toString();
```

**Review by fmorency — APPROVED** [review]
Thanks! Minor comment

- Can you add a test covering edge cases, e.g., numbers bigger than `Number.MAX_SAFE_INTEGER`. We need to make sure those always work.

Relates #331

**Review by fmorency — APPROVED** [review]
